### PR TITLE
Remove last line from users

### PR DESCRIPTION
**Pull request by RafalKorepta**
The following logs output shows that last line shouldn't be taken into users creation:

```
$ kubectl logs -c config-watcher redpanda-1
...
Creating user # intentional empty line...
error creating user # intentional empty line: Error: accepts at most 1 arg(s), received 4
Usage:
  rpk acl user create [USER] -p [PASS] [flags]

Flags:
  -h, --help               Help for create
      --mechanism string   SASL mechanism to use for the user you are
                           creating (scram-sha-256, scram-sha-512, case
                           insensitive) (default "scram-sha-256")
      --password string    New user's password (NOTE: if using --password
                           for the admin API, use --new-password)

Global Flags:
      --config string            Redpanda or rpk config file; default
                                 search paths are ~/.config/rpk/rpk.yaml,
                                 $PWD, and /etc/redpanda/redpanda.yaml
  -X, --config-opt stringArray   Override rpk configuration settings; '-X
                                 help' for detail or '-X list' for terser
                                 detail
      --profile string           rpk profile to use
  -v, --verbose                  Enable verbose logging
```

### CODE REVIEW

**Comment by RafalKorepta** [comment]
Any failure in connectors test should be addressed in https://github.com/redpanda-data/helm-charts/pull/803

**Review by alejandroEsc — APPROVED** [review]
lgtm. as long as the sasl test pass i think we should be free to merge.

**Comment by RafalKorepta** [comment]
In 01-03 (`03-one-node-cluster-tls-no-sasl-values.yaml`) the curl failed to delete connector.

In 04-06 (`04-one-node-cluster-no-tls-sasl-values.yaml`) the backoff in console and connectors reached `--timeout 480s`

The connectors pod status
```
2023-10-17T13:55:59.8126005Z     State:          Waiting
2023-10-17T13:55:59.8126381Z       Reason:       CrashLoopBackOff
2023-10-17T13:55:59.8126796Z     Last State:     Terminated
2023-10-17T13:55:59.8127162Z       Reason:       Error
2023-10-17T13:55:59.8127501Z       Exit Code:    1
2023-10-17T13:55:59.8127876Z       Started:      Tue, 17 Oct 2023 13:53:46 +0000
2023-10-17T13:55:59.8128400Z       Finished:     Tue, 17 Oct 2023 13:53:46 +0000
2023-10-17T13:55:59.8128858Z     Ready:          False
2023-10-17T13:55:59.8129185Z     Restart Count:  6
```

In v23.1.18 04-06 (`04-one-node-cluster-no-tls-sasl-values.yaml`) the backoff in console and connectors reached `--timeout 480s` same as above.

In v22.3.23 04-06 (`04-one-node-cluster-no-tls-sasl-values.yaml`) the backoff in console and connectors reached `--timeout 480s` same as above.

File: charts/redpanda/templates/connectors/connectors.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ limitations under the License.
 {{ $command := list }}
 {{ if (include "sasl-enabled" . | fromJson).bool }}
   {{ $command = concat $command (list "sh" "-c") }}
-  {{ $consoleSASLConfig := (printf "set -e; IFS=: read -r CONNECT_SASL_USERNAME KAFKA_SASL_PASSWORD CONNECT_SASL_MECHANISM < $(find /mnt/users/* -print); CONNECT_SASL_MECHANISM=${CONNECT_SASL_MECHANISM:-%s}; export CONNECT_SASL_USERNAME CONNECT_SASL_PASSWORD_FILE CONNECT_SASL_MECHANISM;" (( include "sasl-mechanism" . ) | lower)) }}
+  {{ $consoleSASLConfig := (printf "set -e; IFS=: read -r CONNECT_SASL_USERNAME KAFKA_SASL_PASSWORD CONNECT_SASL_MECHANISM < <(grep \"\" $(find /mnt/users/* -print)); CONNECT_SASL_MECHANISM=${CONNECT_SASL_MECHANISM:-%s}; export CONNECT_SASL_USERNAME CONNECT_SASL_PASSWORD_FILE CONNECT_SASL_MECHANISM;" (( include "sasl-mechanism" . ) | lower)) }}
```
Comment (RafalKorepta) [inline charts/redpanda/templates/connectors/connectors.yaml:46]: I don't know if that matters, but maybe
```suggestion
  {{ $consoleSASLConfig := (printf "set -e; IFS=':' read -r CONNECT_SASL_USERNAME KAFKA_SASL_PASSWORD CONNECT_SASL_MECHANISM < <(grep \"\" $(find /mnt/users/* -print)); CONNECT_SASL_MECHANISM=${CONNECT_SASL_MECHANISM:-%s}; export CONNECT_SASL_USERNAME CONNECT_SASL_PASSWORD_FILE CONNECT_SASL_MECHANISM;" (( include "sasl-mechanism" . ) | lower)) }}
```

File: charts/redpanda/templates/tests/test-kafka-produce-consume.yaml [hunk 2]
```diff
@@ -59,7 +59,7 @@ spec:
 {{- if .Values.auth.sasl.enabled }}
           old_setting=${-//[^x]/}
           set +x
-          IFS=: read -r {{ include "rpk-sasl-environment-variables" . }} < $(find /etc/secrets/users/* -print)
+          IFS=: read -r {{ include "rpk-sasl-environment-variables" . }} < <(grep \"\" $(find /etc/secrets/users/* -print))
```
Comment (RafalKorepta) [inline charts/redpanda/templates/tests/test-kafka-produce-consume.yaml:62]: Maybe
```suggestion
          IFS=":" read -r {{ include "rpk-sasl-environment-variables" . }} < <(grep \"\" $(find /etc/secrets/users/* -print))
```

File: charts/redpanda/templates/tests/test-kafka-sasl-status.yaml [hunk 3]
```diff
@@ -51,7 +51,7 @@ spec:
 {{- if .Values.auth.sasl.enabled }}
           old_setting=${-//[^x]/}
           set +x
-          IFS=: read -r {{ include "rpk-sasl-environment-variables" . }} < $(find /etc/secrets/users/* -print)
+          IFS=: read -r {{ include "rpk-sasl-environment-variables" . }} < <(grep \"\" $(find /etc/secrets/users/* -print))
```
Comment (RafalKorepta) [inline charts/redpanda/templates/tests/test-kafka-sasl-status.yaml:54]: Maybe
```suggestion
          IFS=":" read -r {{ include "rpk-sasl-environment-variables" . }} < <(grep \"\" $(find /etc/secrets/users/* -print))
```

**Review by RafalKorepta — COMMENTED** [review]
Rest tests have the same `IFS`, but I'm not sure if semicolon needs to be quoted.

File: charts/redpanda/templates/connectors/connectors.yaml (hunk 1 above)
Comment (alejandroEsc) [reply charts/redpanda/templates/connectors/connectors.yaml:46]: It doesnt, but ill commit it

**Pull request merged**